docs/sentry/react-native.md

```diff
@@ -0,0 +1,84 @@
+# Sentry React Native
+
+
+### Getting Started
+Install all needed dependencies:
+* [@sentry/react-native](https://www.npmjs.com/package/@sentry/react-native)
+
+```bash
+npm i @sentry/react-native
+#or
+yarn add @sentry/react-native
+```
+
+Run the Sentry wizard to automatically patch native files. (Hint: The wizard doesn't support React Native versions 0.68 and older)  
+**Remember to work on a clean workspace and commit current changes. So in case of failure you don't mess up your code.**
+```bash
+npx @sentry/wizard@latest -s -i reactNative
+```
+
+The wizard creates two sentry.properties files. One in the android and one in the ios native directories.  
+Add `sentry.properties` to `.gitignore` as it contains the secret access token for your CI.  
+You can copy the contents of this file as base64 and add it as a secret in your workflow.
+```bash
+# This copies the file to your clipboard
+cat ios/sentry.properties | openssl base64 | tr -d '\n' | pbcopy
+```
+Add the base64 string as a secret to your github repository like `SENTRY_PROPERTIES_FILE_BASE64` with the content of your clipboard  
+In your CI you can use something like this to recreate the file
+```yaml
+- name: Add sentry.properties files
+  run: |
+    echo "$SENTRY_PROPERTIES_FILE_BASE64" | base64 --decode > ios/sentry.properties
+    echo "$SENTRY_PROPERTIES_FILE_BASE64" | base64 --decode > android/sentry.properties
+```
+
+For monorepos or when your react-native application is not present in your repository root you have to make a slight change to your ios `project.pbxproj` file.  
+Search for the `Bundle React Native code and images` task and add `export PROJECT_ROOT=".."` to the top of the shellScript command. Right before the `export SENTRY_PROPERTIES...`
+
+
+### Setting up the SDK
+The previous steps should have set the stage to actually use the sentry sdk.  
+
+First you need to initialize the SDK like so:
+
+```ts
+// you could write a wrapperservice like a trackingservice which handles this for you.
+Sentry.init({
+  // you don't need to make your dsn string a secret.
+  dsn: 'your_dsn_string',
```

Comment:
Is this string sensitive? Do we have to set it as env or can we leave it as string in VCS?

Reply:
I am not 100% sure, but as far as i know it is not sensitive. will look into this today. But because it is in the app bundle it cant be secured entirely.

Reply:
Sentry also tracks console logs (log, info, warning, error) by default and therefore kind serves as log rotation tool
These are safe to store in vcs